error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ImmutableSetRules.java

```diff
@@ -211,4 +214,43 @@ ImmutableSet<T> after(T e1, T e2, T e3, T e4, T e5) {
       return ImmutableSet.of(e1, e2, e3, e4, e5);
     }
   }
+
+  /** Prefer {@link Sets#intersection(Set, Set)} over more contrived alternatives. */
+  static final class SetsIntersection<T> {
+    @BeforeTemplate
+    ImmutableSet<T> before(Set<T> set1, Set<T> set2) {
+      return set1.stream().filter(set2::contains).collect(toImmutableSet());
+    }
```

Comment:
Here and below: the sets don't need to be of the same type `T` :)

Reply:
Will sets of same type still be refastered after this ?

Reply:
yep, it matches for any `S` and `T`, so also of `S == T`.